zk/lock.go

```diff
@@ -146,5 +157,46 @@ func (l *Lock) Unlock() error {
 	}
 	l.lockPath = ""
 	l.seq = 0
+	delete(mapEphermeralSequenceLockPath,l.path)
 	return nil
 }
+
+//Checks whether lock got created and response was lost because of network partition failure.
+//It queries zookeeper and scans existing sequential ephemeral znodes under the parent path
+//It finds out that previously requested sequence number corresponds to child having lowest sequence number
+func lockExists(c *Conn, rootPath string, znodePath string) bool {
+	seq, err := parseSeq(znodePath)
+	if err != nil {
+		return false
+	}
+
+	//scans the existing znodes if there are any
+	for {
```

Comment:
as far as i see this for loop is redundant as in the end of it you aslways return `true` or `false` in any way. is that intended? or you wanted your logic to work in other way?

Reply:
You are absolutely spot on. I somehow ignored it when I reused the existing loop.